src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py

```diff
@@ -207,42 +211,54 @@ def set_user_admin(self, user_object_id, **kwargs):
     def create_aad_user(self):
         query_list = self.get_create_query()
         connection_kwargs = self.get_connection_string()
-        ip_name = None
+        ip_name = generate_random_string(prefix='svc_').lower()
         try:
             logger.warning("Connecting to database...")
             self.create_aad_user_in_mysql(connection_kwargs, query_list)
-        except AzureConnectionError:
-            # allow public access
-            ip_name = generate_random_string(prefix='svc_').lower()
-            self.set_target_firewall(True, ip_name)
+        except AzureConnectionError as e:
+            logger.warning(e)
+            # allow local access
+            from requests import get
+            ip_address = get(IP_ADDRESS_CHECKER).text
+            self.set_target_firewall(True, ip_name, ip_address, ip_address)
             # create again
-            self.create_aad_user_in_mysql(connection_kwargs, query_list)
-
-        # remove firewall rule
-        if ip_name is not None:
             try:
+                self.create_aad_user_in_mysql(connection_kwargs, query_list)
+            except AzureConnectionError as e:
+                logger.warning(e)
+                # allow public access
+                self.set_target_firewall(True, ip_name, '0.0.0.0', '255.255.255.255')
```

Comment:
need user confirmation on this, how to add a confirmation option here?

Reply:
Updated